lib/connection_controller.go

```diff
@@ -141,13 +203,34 @@ func (cc *ConnectionController) cleanupFailedOutboundConnection(connection Conne
 	if rn != nil {
 		cc.rnManager.Disconnect(rn)
 	}
+	oc.Close()
 	cc.cmgr.RemoveAttemptedOutboundAddrs(oc.address)
 }
 
 // ###########################
-// ## Connections
+// ## Persistent Connections
 // ###########################
 
+func (cc *ConnectionController) refreshConnectIps() {
+	// Connect to addresses passed via the --connect-ips flag. These addresses are persistent in the sense that if we
+	// disconnect from one, we will try to reconnect to the same one.
+	for _, connectIp := range cc.connectIps {
+		if _, ok := cc.persistentIpToRemoteNodeIdsMap[connectIp]; ok {
+			continue
+		}
+
+		glog.Infof("ConnectionController.initiatePersistentConnections: Connecting to connectIp: %v", connectIp)
+		id, err := cc.CreateNonValidatorPersistentOutboundConnection(connectIp)
```

Comment:
so connect-ips flag won't allow a connection to a validator node?

Reply:
Persistent remote nodes can still be validators. The ValidatorConnector routine may reindex the persistent remote node after completing the handshake. We default to a NonValidator connection so that we don't have to assume the public key of this RemoteNode (Validator RemoteNodes require an expected public key passed on initialization).